core/src/test/scala/unit/kafka/admin/ResetConsumerGroupOffsetTest.scala

```diff
@@ -109,9 +109,9 @@ class ResetConsumerGroupOffsetTest extends ConsumerGroupCommandTest {
     val args = buildArgsForGroup(group, "--all-topics", "--to-current", "--execute")
     val consumerGroupCommand = getConsumerGroupService(args)
     // Make sure we got a coordinator
-    TestUtils.waitUntilTrue(() => {
+    TestUtils.waitUntilTrue(() => TestUtils.maybeUnwrapFutureException(() => {
       consumerGroupCommand.collectGroupState(group).coordinator.host() == "localhost"
-    }, "Can't find a coordinator")
+    }), "Can't find a coordinator", maxRetries = 3)
```

Comment:
How did you choose the number 3? One thing I noticed is that `waitUntilTrue` does not have a backoff between retries in the case of an exception. Perhaps it should? 

I think in these cases our expectation is actually that we will keep retrying until the timeout is reached. I'm wondering if we can get rid of `maxRetries` and just use a boolean flag which says whether or not to retry on retriable errors.

Reply:
Yeah, that sounds like a better approach. I've replaced the count with a boolean and added a pause before retrying.